### PR TITLE
Main menu

### PR DESCRIPTION
**Pull request by mjhult**
## Completed
- Created rough custom focus handler.
- Added MainMenu
- Added buttons to MainMenu
- Added children to MainMenu
- Injected custom isFocused prop to MainMenu children
- Modified PlaybackControls and VolumeControl to use injected isFocused prop.
- Slight refactoring
- Removed required "enter" key press before volume change.

## In progress
- Testing

## TODO
- More in depth refactoring
- MainMenu button wrap around on side arrow press

### CODE REVIEW

**Comment by mjhult** [comment]
Moved loadingResults to SearchResults due to a bug where the SearchResults component would never be invoked resulting in infinite "Loading Results" method.

**Comment by mjhult** [comment]
Added testState prop to MainMenu to bypass lack of simulated keyboard events to test child and button focusing.

File: src/App.tsx [hunk 1]
```diff
@@ -43,16 +42,23 @@ export const App: React.FC<AppProps> = ({
     };
   }, []);
 
+  // TODO: Find better place for MainMenu items to be decided.
+  const menuItems: MenuButtonItem[] = [
+    { label: 'Search 🔎', element: <SearchBar />, action: (id) =>  {} },
+    { label: 'Exit 🛑', action: () => process.exit(1) },
+  ];
```
Comment (WVAviator) [inline src/App.tsx:49]: Let's just remove the emojis for now. Maybe we'll figure out another solution later.

File: src/components/MainMenu/MainMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+                         flexDirection='row'
+                        >
+                        { selected.buttonFocused && selected.buttonId === j ? items[j].element ?? <></> : <Text>{`${items[j].label}`}</Text> }
+                    </Box>
+                );
+            }
+            return (
+                <Box width='100%'
+                     key={`row-${i}`} 
+                     justifyContent='center'
```
Comment (WVAviator) [inline src/components/MainMenu/MainMenu.tsx]: remove this to align the boxes to the left

**Review by WVAviator — APPROVED** [review]

**Pull request merged**